numpyro/patch.py

```diff
@@ -15,3 +18,9 @@ def decorator(new_fn):
         return new_fn
 
     return decorator
+
+
+@patch_dependency('jax.api._check_inexact_input_vjp', jax)
```

Comment:
I am a bit hesitant to patch over and bypass this check, as it might lead to hard to diagnose bugs in the future. I also think that the JAX team probably didn't intend to expose `ravel_pytree` for external usage. How about we just define our own `ravel_pytree` in `numpyro.util` and not use `vjp`? 

```python
def ravel_list(*leaves):
    ravel_leaves = [np.ravel(elt) for elt in leaves] if leaves else [np.array([])]
    len_leaves = [len(l) for l in ravel_leaves]
    unravel_list = lambda arr: [arr[i: i+l] for i, l in enumerate(len_leaves)]
    return np.concatenate(ravel_leaves), unravel_list


def ravel_pytree(pytree):
    leaves, treedef = tree_flatten(pytree)
    flat, unravel_list = ravel_list(*leaves)
    unravel_pytree = lambda flat: tree_unflatten(treedef, unravel_list(flat))
    return flat, unravel_pytree
```

We can also cache that if it is being called frequently, though I'm not sure if we need to do that.

Reply:
@neerajprad Could you make a PR for your solution? Your code sounds reasonable to me and I agree that we should go for a solution like yours.

Reply:
Sure, I'll send in a PR shortly.

Reply:
Well my snippet above isn't entirely correct, and this turned out to be more complex, e.g. when we have mixed types - bools, ints, etc. I should have something up soon though.